sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -39,19 +39,31 @@ public static HttpPipeline Build(ClientOptions options, params HttpPipelinePolic
             HttpPipelinePolicy[] perRetryPolicies,
             ResponseClassifier? responseClassifier)
         {
-            return Build(options, perCallPolicies, perRetryPolicies, responseClassifier, null);
+            var result = BuildInternal(options, perCallPolicies, perRetryPolicies, null, responseClassifier);
+            return new HttpPipeline(result.Transport, result.PerCallIndex, result.PerRetryIndex, result.Policies.ToArray(), result.Classifier);
         }
 
         /// <summary>
-        /// Creates an instance of <see cref="HttpPipeline"/> populated with default policies, customer provided policies from <paramref name="options"/> and client provided per call policies.
+        /// Creates an instance of <see cref="DisposableHttpPipeline"/> populated with default policies, customer provided policies from <paramref name="options"/> and client provided per call policies.
```

Comment:
Is it worth changing the text of this comment to describe the differences between these two methods?  When would I want to call one vs. the other?

Reply:
I think an updated description here may address the naming issue for the defaultTransportOptions.